docs/source/community/contributing.mdx

```diff
@@ -0,0 +1,116 @@
+
+To support a new model architecture in the Optimum Neuron library here are some steps to follow:
+
+1. Implement a custom Neuron configuration.
+2. Export and validate the model.
+3. Contribute to the GitHub repo. 
+
+## Implement a custom Neuron configuration
+
+To support the export of a new model to a Neuron compatible format, the first thing to do is to define a Neuron configuration, describing how to export the PyTorch model by specifying:
+
+1. The input names.
+2. The output names.
+3. The dummy inputs used to trace the model. (The Neuron Compiler asks this to record the computational graph and embed it to a TorchScript module.)
+4. The compilation arguments used to control the trade-off between hardware efficiency (latency, throughput) and accuracy.
+
+Depending on the choice of model and task, we represent the data above with configuration classes. Each configuration class is associated with
+a specific model architecture, and follows the naming convention `ArchitectureNameNeuronConfig`. For instance, the configuration that specifies the Neuron
+export of BERT models is `BertNeuronConfig`.
+
+Since many architectures share similar properties for their Neuron configuration, 🤗 Optimum adopts a 3-level class hierarchy:
+
+1. Abstract and generic base classes. These handle all the fundamental features, while being agnostic to the modality (text, image, audio, etc).
+2. Middle-end classes. These are aware of the modality. Multiple config classes could exist for the same modality, depending on the inputs they support. They specify which input generators should be used for generating the dummy inputs, but remain model-agnostic.
+3. Model-specific classes like the `BertNeuronConfig` mentioned above. These are the ones actually used to export models.
+
+### Example: Adding support for ESM models
+
+Here we take the support of [ESM models](https://huggingface.co/docs/transformers/model_doc/esm#esm) as an example. Let's create an `EsmNeuronConfig` class in the `optimum/exporters/neuron/model_configs.py`.
+
+When an Esm model interprets as a text encoder, we are able to inherit from the middle-end class [`TextEncoderNeuronConfig`](https://github.com/huggingface/optimum-neuron/blob/v0.0.18/optimum/exporters/neuron/config.py#L36). 
+Since the modeling and configuration of Esm is almost the same as BERT when it is interpreted as an encoder, we can use the `NormalizedConfigManager` with `model_type=bert` to normalize the configuration to generate dummy inputs for tracing the model.
+
+And one last step, since `optimum-neuron` is an extension of `optimum`, we need to register the Neuron config that we create to the [TasksManager](https://huggingface.co/docs/optimum/main/en/exporters/task_manager#optimum.exporters.TasksManager) with the `register_in_tasks_manager` by specifying the model type and supported tasks.
```

Comment:
```suggestion
And one last step, since `optimum-neuron` is an extension of `optimum`, we need to register the Neuron config that we create to the [TasksManager](https://huggingface.co/docs/optimum/main/en/exporters/task_manager#optimum.exporters.TasksManager) with the `register_in_tasks_manager` decorator by specifying the model type and supported tasks.
```